src/lm/tools/summarizeNotificationsTool.ts

```diff
@@ -44,32 +49,34 @@ Patch: ${fileChange.patch}
 			}
 		}
 
-		const unreadComments = options.parameters.unreadComments;
-		if (unreadComments.length > 0) {
+		const unreadComments = options.parameters.comments;
+		if (unreadComments && unreadComments.length > 0) {
 			notificationInfo += `
 The following are the unread comments of the thread:
 `;
-		}
-		for (const [index, comment] of unreadComments.entries()) {
-			notificationInfo += `
+			for (const [index, comment] of unreadComments.entries()) {
+				notificationInfo += `
 Comment ${index} :
 Author: ${comment.author}
 Body: ${comment.body}
 `;
+			}
 		}
 		const models = await vscode.lm.selectChatModels({
 			vendor: 'copilot',
 			family: 'gpt-4o'
 		});
 		const model = models[0];
-		const markAsReadCommand: vscode.Command = {
-			title: 'Mark As Read',
-			command: 'notification.markAsRead',
-			arguments: [{
-				threadId: options.parameters.threadId,
-				notificationKey: options.parameters.notificationKey
-			}]
-		};
+		let markAsReadCommand: vscode.Command | undefined;
+		const threadId = options.parameters.threadId;
+		const notificationKey = options.parameters.notificationKey;
+		if (threadId && notificationKey) {
+			markAsReadCommand = {
```

Comment:
If `markAsReadCommand` does not end up getting defined here, then below we shouldn't add it into the `LanguageModelToolResult` that we return.

Reply:
That's right! Thanks for picking up on that. I'll change that